app/models/search/work_query.rb

```diff
@@ -275,20 +275,29 @@ def facet_tags?
     options[:faceted]
   end
 
-  def facet_collections?
+  def collected?
     options[:collected]
   end
 
   def include_restricted?
     User.current_user.present? || options[:show_restricted]
   end
 
+  # Include unrevealed works only if we're on a collection page
+  # OR the collected works page of a user
   def include_unrevealed?
-    options[:collection_ids].present?
+    options[:collection_ids].present? || collected?
   end
 
+  # Include anonymous works unless we're on a user or pseud page
+  # OR unless the user is viewing their own collected works
```

Comment:
I think this part of the comment is backwards -- we're including anonymous works _if_ the user is viewing their own collected works page, not _unless_. It'd be "OR unless we're viewing someone else's collected works"